crates/wdl-lint/src/rules/parameter_description.rs

```diff
@@ -0,0 +1,185 @@
+//! A lint rule for ensuring parameters have proper descriptions.
+
+use wdl_analysis::Diagnostics;
+use wdl_analysis::VisitReason;
+use wdl_analysis::Visitor;
+use wdl_ast::AstNode;
+use wdl_ast::AstToken;
+use wdl_ast::Diagnostic;
+use wdl_ast::Span;
+use wdl_ast::SyntaxElement;
+use wdl_ast::SyntaxKind;
+use wdl_ast::v1::MetadataSection;
+use wdl_ast::v1::MetadataValue;
+use wdl_ast::v1::ParameterMetadataSection;
+
+use crate::Rule;
+use crate::Tag;
+use crate::TagSet;
+
+/// The identifier for the parameter description rule.
+const ID: &str = "ParameterDescription";
+
+/// The reserved key name for descriptions.
+const DESCRIPTION_KEY: &str = "description";
+
+/// Creates a diagnostic for missing descriptions.
+fn missing_description_diagnostic(name: &str, is_output: bool, span: Span) -> Diagnostic {
+    let item_type = if is_output { "output" } else { "parameter" };
+    let location = if is_output { " in `meta.outputs`" } else { "" };
+
+    Diagnostic::warning(format!(
+        "{} `{}` is missing a description{}",
+        item_type, name, location
+    ))
+    .with_rule(ID)
+    .with_label(
+        format!(
+            "{} should be documented with either a `String` description or an object with a \
+             `description` key",
+            item_type
+        ),
+        span,
+    )
+    .with_fix(format!(
+        "add a description for `{}` (e.g., `{}: \"description\"` or `{}: {{ description: \
+         \"description\" }}`)",
```

Comment:
```suggestion
        "add a description for `{}`",
```
nit: I think keeping things simple here is better than complicating the message